Frameworks/UIKit/UIButton.mm

```diff
@@ -503,44 +545,34 @@ - (UIImage*)currentBackgroundImage {
 - (void)setTitle:(NSString*)title forState:(UIControlState)state {
     _states[state].title = [title copy];
 
-    // NOTE: check if title is nil before creating inspectableTitle
-    // createString:nil creates a valid rtString with null comObj
-    // which isn't what we want
-    if (title) {
-        RTObject* rtString = [WFPropertyValue createString:title];
-        if (rtString) {
-            _states[state].inspectableTitle = [rtString comObj];
-        }
-    } else {
-        // this enforces the fallback of using title of normalState
-        // when a title for other states does not exist
-        _states[state].inspectableTitle = nullptr;
-    }
-
-    // Update the Xaml elements immediately, so the proxies reflect reality
-    XamlButtonApplyVisuals([_xamlButton comObj],
-                           _currentInspectableTitle(self),
-                           _currentInspectableImage(self),
-                           _currentInspectableTitleColor(self));
+    // Update our title label immediately
+    self.titleLabel.text = self.currentTitle;
```

Comment:
got you, setting value on other states might impact current state title, you need re-evaluate.

---
In reply to: [100653762](https://github.com/Microsoft/WinObjC/pull/1920#discussion_r100653762) [](ancestors = 100653762,100653633,100653158,100649772)